sdk/events.go

```diff
@@ -14,6 +21,77 @@ type PushEvent struct {
 	}
 	AfterCommitID string `json:"after"`
 	Installation  struct {
-		Id int `json:"id"`
+		ID int `json:"id"`
+	}
+}
+
+// EventInfo to pass/store events across functions
+type Event struct {
+	Service        string            `json:"service"`
+	Owner          string            `json:"owner"`
+	Repository     string            `json:"repository"`
+	Image          string            `json:"image"`
+	Sha            string            `json:"sha"`
+	URL            string            `json:"url"`
+	InstallationID int               `json:"installationID"`
+	Environment    map[string]string `json:"environment"`
+	Secrets        []string          `json:"secrets"`
+}
+
+// function to build Event from PushEvent
+func BuildEventFromPushEvent(pushEvent PushEvent) *Event {
+	info := Event{}
+
+	info.Service = pushEvent.Repository.Name
+	info.Owner = pushEvent.Repository.Owner.Login
+	info.Repository = pushEvent.Repository.Name
+	info.Sha = pushEvent.AfterCommitID
+	info.URL = pushEvent.Repository.CloneURL
+	info.InstallationID = pushEvent.Installation.ID
+
+	return &info
+}
+
+// function to build Event from Environment Variable
+func BuildEventFromEnv() (*Event, error) {
```

Comment:
I don't think this belongs in the status package, this is not generic because the HTTP headers may be different between functions. This should remain in the git-tar function where it is used. It is also only called once which shows it should remain in its own package (the function)

Reply:
This is not additionally added for `status` package but it is defined in package `sdk` at `event.go`. And its being used from `buildshiprun`  not `git-tar`